### PR TITLE
Fixes #5685 add provider calendar methods

### LINKED ISSUE #5685: Add provider calendar methods into UserService
Right now to get a list of providers who have calendar access the function for that is inside patient.inc inside getProviderInfo() inside a nested if condition.  Rather than module writers having to duplicate that code we should add to our UserService class methods to grab a list of calendar users with the option to filter by facility.  This makes it easier for core code or modules to grab the calendar users in other areas of OpenEMR.

### PR DESCRIPTION
**Pull request by adunsulag**
Add methods to the user service and have the calendar api use those methods to grab the calendar providers.

### CODE REVIEW

**Review by bradymiller — APPROVED** [review]

File: src/Services/UserService.php [hunk 1]
```diff
@@ -137,6 +137,69 @@ public function getUserByUUID($uuid)
...
+        //--------------------------------
+        //(CHEMED) facility filter
+        $param2 = "";
+        if (!empty($facility)) {
+            if ($GLOBALS['restrict_user_facility']) {
+                // so is facility a string... or is it a column name... this is very confusing here
+                $param2 = " AND (facility_id = '" . add_escape_custom($facility) . "' OR  '" . add_escape_custom($facility)
+                    . "' IN (select facility_id from users_facility where tablename = 'users' and table_id = id))";
+            } else {
+                $param2 = " AND facility_id = '" . add_escape_custom($facility) . "' ";
```
Comment (bradymiller) [inline src/Services/UserService.php:181]: Could consider improving this part of the query by using binding (although not sure if the use of binding will work in that facility after the `OR`, although would be easy to test).

File: src/Services/UserService.php [hunk 2]
```diff
@@ -137,6 +137,69 @@ public function getUserByUUID($uuid)
...
+        return $this->searchUsersForCalendar("", $userId);
+    }
+
+    /**
+     * Retrieves all the users that have been set to show up on the calendar optionally filtered by the facility if one
+     * is provided.
+     * @param string $facility
+     * @return array|null
+     */
+    public function getUsersForCalender($facility = "")
```
Comment (stephenwaite) [inline src/Services/UserService.php:157]: minor spelling of Calender vs Calendar but is consistent above

Comment (adunsulag) [reply src/Services/UserService.php]: Thanks for the catch!

File: src/Services/UserService.php (hunk 1 above)
Comment (adunsulag) [reply src/Services/UserService.php]: I changed it to use binding and it appears to work fine with what I've tested.

**Pull request merged**